Sources/Soto/Extensions/DynamoDB/DynamoDB+Codable+async.swift

```diff
@@ -0,0 +1,150 @@
+    ///
+    /// For information on how to call the `PutItem` API using the AWS SDK in specific languages, see the following:</p> <ul> <li> <p> <a href="http://docs.aws.amazon.com/goto/aws-cli/dynamodb-2012-08-10/PutItem"> PutItem in the AWS Command Line Interface</a> </p> </li> <li> <p> <a href="http://docs.aws.amazon.com/goto/DotNetSDKV3/dynamodb-2012-08-10/PutItem"> PutItem in the AWS SDK for .NET</a> </p> </li> <li> <p> <a href="http://docs.aws.amazon.com/goto/SdkForCpp/dynamodb-2012-08-10/PutItem"> PutItem in the AWS SDK for C++</a> </p> </li> <li> <p> <a href="http://docs.aws.amazon.com/goto/SdkForGoV1/dynamodb-2012-08-10/PutItem"> PutItem in the AWS SDK for Go</a> </p> </li> <li> <p> <a href="http://docs.aws.amazon.com/goto/SdkForJava/dynamodb-2012-08-10/PutItem"> PutItem in the AWS SDK for Java</a> </p> </li> <li> <p> <a href="http://docs.aws.amazon.com/goto/AWSJavaScriptSDK/dynamodb-2012-08-10/PutItem"> PutItem in the AWS SDK for JavaScript</a> </p> </li> <li> <p> <a href="http://docs.aws.amazon.com/goto/SdkForPHPV3/dynamodb-2012-08-10/PutItem"> PutItem in the AWS SDK for PHP V3</a> </p> </li> <li> <p> <a href="http://docs.aws.amazon.com/goto/boto3/dynamodb-2012-08-10/PutItem"> PutItem in the AWS SDK for Python</a> </p> </li> <li> <p> <a href="http://docs.aws.amazon.com/goto/SdkForRubyV2/dynamodb-2012-08-10/PutItem"> PutItem in the AWS SDK for Ruby V2</a> </p> </li> </ul> </important> <p>When you add an item, the primary key attributes are the only required attributes. Attribute values cannot be null.
+    ///
+    /// Empty String and Binary attribute values are allowed. Attribute values of type String and Binary must have a length greater than zero if the attribute is used as a key attribute for a table or index. Set type attributes cannot be empty.
+    ///
+    /// Invalid Requests with empty values will be rejected with a `ValidationException` exception.</p> <note> <p>To prevent a new item from replacing an existing item, use a conditional expression that contains the `attribute_not_exists` function with the name of the attribute being used as the partition key for the table. Since every record must contain that attribute, the `attribute_not_exists` function will only succeed if no matching item exists.</p> </note> <p>For more information about `PutItem`, see <a href="https://docs.aws.amazon.com/amazondynamodb/latest/developerguide/WorkingWithItems.html">Working with Items</a> in the <i>Amazon DynamoDB Developer Guide</i>.
+    public func putItem<T: Encodable>(
+        _ input: PutItemCodableInput<T>,
+        logger: Logger = AWSClient.loggingDisabled,
+        on eventLoop: EventLoop? = nil
+    ) async throws -> PutItemOutput {
+        try await self.putItem(input, logger: logger, on: eventLoop).get()
```

Comment:
We should use the `async` version of `putItem` instead of the ELF version to make the transition easier longer term IMO. Same goes for everything in this file

Reply:
Fair point. Will fix. I should probably do something similar for S3.multipart but that is a fair amount of work so will do in another PR

Reply:
Just had a look at this and this would mean replicating the code in DynamoDB+Codable is that what you intended?

Reply:
Added them in so you can see changes

Reply:
Ah right, no that's not quite what I intended actually. I didn't realise it was going to involve that. However, I definitely think it's the longer term solution, but agree that it's probably overkill for each service addition until we actually need it

Reply:
(Other option of course is to extract all the code that's duplicated to avoid duplicating all the logic)

Reply:
> (Other option of course is to extract all the code that's duplicated to avoid duplicating all the logic)

Did as you said.
Also added some tests to ensure it all works